cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/connection/ConnectorDetail.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.etl.proto.connection;
+
+import io.cdap.cdap.api.data.schema.Schema;
+
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Response for the spec endpoint. The schema and properties are set on each available plugins.
+ * This looks duplicated but it is our standard way on representing a plugin.
+ * Detail information about a connector, contains all the plugins related to the connector and their versions.
+ */
+public class ConnectorDetail {
+  private final Set<PluginDetail> relatedPlugins;
+
+  public ConnectorDetail(Set<PluginDetail> relatedPlugins) {
+    this.relatedPlugins = relatedPlugins;
```

Comment:
so the plugin to connector relationship is no longer retrieved from metadata ?

Reply:
It is still retrieved from metadata if you want to know relation from source. Connector is not guaranteed to only relate to one source, so metadata properties might not be enough to represent them. But source is guaranteed to be related to one connector so metadata is good.